### PR TITLE
dist/ethos: fixed compile warnings

### PR DESCRIPTION
**Pull request by haukepetersen**
- checking return values of each write() call
- chack return value of system() call

### CODE REVIEW

File: dist/tools/ethos/ethos.c [hunk 1]
```diff
@@ -33,6 +33,13 @@ static void usage(void)
     fprintf(stderr, "usage: ethos <tap> <serial>\n");
 }
 
+static void checked_write(int handle, void *buffer, int nbyte)
+{
+    if (write(handle, buffer, nbyte) != nbyte) {
+        fprintf(stderr, "write to fd %i failed\n", handle);
```
Comment (kaspar030) [inline dist/tools/ethos/ethos.c]: ```fprintf(stderr, "write to fd %i failed: %s\n", handle, strerror(errno)));```

**Comment by kaspar030** [comment]
Much better. I you feel like, add the actual error message to the output (see other comment). ACK.

**Comment by haukepetersen** [comment]
fixed write error message and squashed.

**Comment by haukepetersen** [comment]
just came to mind when setting the 'read for...' label: tools like ethos are not checked by the CI in any case, right? Does it make sense to include them (open an issue for that)? I think it might not be too bad to do this...

**Comment by kaspar030** [comment]
a ")" too much:

```
[kaspar@booze ethos (test)]$ make
cc -O3 -Wall ethos.c -o ethos
ethos.c: In function ‘checked_write’:
ethos.c:39:80: error: expected ‘;’ before ‘)’ token
         fprintf(stderr, "write to fd %i failed: %s\n", handle, strerror(errno)));
                                                                                ^
ethos.c:39:80: error: expected statement before ‘)’ token
Makefile:4: recipe for target 'ethos' failed
make: *** [ethos] Error 1
[kaspar@booze ethos (test)]$
```

**Comment by kaspar030** [comment]
> tools like ethos are not checked by the CI in any case, right? Does it make sense to include them (open an issue for that)? I think it might not be too bad to do this...

yes and yes...

**Comment by haukepetersen** [comment]
done #5319

**Comment by haukepetersen** [comment]
so are you bold enough to just merge this?

**Comment by kaspar030** [comment]
Did you fix the bracket?

**Comment by kaspar030** [comment]
https://github.com/haukepetersen/RIOT/pull/27

**Comment by haukepetersen** [comment]
nope, github did not update the view before my comments, so I did not see your input.

**Comment by haukepetersen** [comment]
squashed

**Comment by kaspar030** [comment]
compiled & tested here, works like a charm. skipping murdock as this doesn't get compiled anywasy. go!

**Comment by kaspar030** [comment]
Thanks @haukepetersen for taking the time to clean up my sloppyness!

**Comment by haukepetersen** [comment]
my pleasure :-)

**Comment by cgundogan** [comment]
does this need backporting?

**Comment by haukepetersen** [comment]
yes

**Pull request merged**